main.go

```diff
@@ -101,37 +107,47 @@ func main() {
 		// loop indefinitely
 		for {
 			log.Info().Msg("Watching services for all namespaces...")
-			watcher, err := client.CoreV1().WatchServices(context.Background(), k8s.AllNamespaces)
+			timeoutSeconds := int64(300)
+			watcher, err := kubeClientset.CoreV1().Services("").Watch(metav1.ListOptions{
+				TimeoutSeconds: &timeoutSeconds,
+			})
```

Comment:
in letsencrypt controller there is: `defer watcher.Close()` right after creation.
Should we do something similar for this? Maybe close the watcher's result channel

Reply:
This watcher no longer has a `Close` method.